iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorNonLinearModelImpl.java

```diff
@@ -49,19 +49,46 @@ public double getG() {
 
     @Override
     public double getMaximumB() {
-        return sections.values().stream()
-                .map(SectionImpl::getB)
-                .max(Double::compare)
-                .orElseThrow(() -> new PowsyblException("a shunt compensator must have at least one section"));
+        if (sections.values().stream().allMatch(section -> section.getB() >= 0)) {
+            return sections.values().stream().mapToDouble(SectionImpl::getB).sum();
+        } else if (sections.values().stream().allMatch(section -> section.getB() <= 0)) {
+            return 0;
+        }
+        return sections.keySet().stream().mapToDouble(this::getCurrentB).max().orElseThrow(() -> new PowsyblException("a shunt compensator must have at least one section"));
     }
 
     @Override
     public double getMaximumG() {
+        double maxG = sections.keySet().stream()
+                .mapToDouble(this::getCurrentG)
+                .filter(g -> !Double.isNaN(g))
+                .max()
+                .orElse(Double.NaN);
+        if (!Double.isNaN(maxG)) {
+            return Math.max(0, maxG);
+        }
+        return maxG;
+    }
+
+    @Override
+    public double getMinimumB() {
         return sections.values().stream()
-                .map(SectionImpl::getG)
+                .mapToDouble(SectionImpl::getB)
+                .filter(b -> b <= 0)
+                .sum();
+    }
```

Comment:
This is exactly the same algorithm for getMaximumB()?

Reply:
It is the sum of negative `b` (and if there is none, it is zero)